src/ripple/overlay/impl/Manifest.cpp

```diff
@@ -275,6 +276,17 @@ ManifestDisposition
 ManifestCache::applyManifest (
     Manifest m, ValidatorList& unl, beast::Journal journal)
 {
+    /*
+        Move master public key from permanent trusted key list
+        to manifest cache.
+    */
+    if (unl.trusted(m.masterKey))
+    {
+        auto comment = unl.member (m.masterKey);
+        addTrustedKey (m.masterKey, comment.get());
+        unl.removePermanentKey (m.masterKey);
+    }
```

Comment:
This is even simpler and more elegant than I expected. I'll note that if the given key is in *both* sections, `addTrustedKey` will `throw`, but, because the same type of check and result will happen in `loadValidatorKeys` if a key is duplicated in a single section, this is almost certainly intentional.  The only _downside_ is that the `loadValidatorKeys` check happens at startup, and this one happens at later runtime. Not sure if that needs to be compensated for.